clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1198,18 +1209,22 @@ boolean canBumpEpoch() {
         return coordinatorSupportsBumpingEpoch;
     }
 
+    private void resetTransactions() {
+        newPartitionsInTransaction.clear();
+        pendingPartitionsInTransaction.clear();
+        partitionsInTransaction.clear();
+    }
+
     private void completeTransaction() {
         if (epochBumpRequired) {
             transitionTo(State.INITIALIZING);
         } else {
             transitionTo(State.READY);
         }
         lastError = null;
+        abortableError = null;
         epochBumpRequired = false;
-        transactionStarted = false;
```

Comment:
Do we miss the update for this flag?

Reply:
Thanks for reminding.

Reply:
If this is the case, could we just remove this intermediate variable?
================================
It could have been omitted, but now this variable has a new use when call initTransactions.